compiler/rustc_hir_analysis/src/check/intrinsic.rs

```diff
@@ -473,6 +474,11 @@ pub fn check_intrinsic_type(tcx: TyCtxt<'_>, it: &hir::ForeignItem<'_>) {
 
             sym::black_box => (1, vec![param(0)], param(0)),
 
+            sym::is_constant => {
+                // FIXME: ZSTs cause an ICE. We should check for this.
```

Comment:
Ideally the codegen backend should be fixed to not ICE. :)

Reply:
We have lots of intrinsics that ICE if used wrongly. Seems fine to me to just document "don't use this on zsts" for now

Reply:
Note that per https://github.com/rust-lang/compiler-team/issues/620, intrinsics ICEing when misused is totally fine.

I might just say to remove the FIXME, but document on the intrinsic that it shouldn't be expected to work for anything but non-ZST scalar primitives.